### ISSUE TITLE
Handymen cut grass incorrectly

### ISSUE THREAD

**Issue by Krutonium** [OpenRCT2/OpenRCT2#10292]
**OS:** [Arch Linux
**Version:** 0.2.4-95
**Commit/Build:** 40e22eba8

Handymen cut the grass one way, and the cut grass is rendered in a way that indicates they mowed in a different direction

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. Place Handyman on uncut grass
2. Allow him to cut grass

**Dump file**

**Screenshots / Video:**

**Save game:**

**Comment by tupaschoal** [comment]
Behavior on:
- https://github.com/OpenRCT2/OpenRCT2/commit/073e4b0cc: https://i.imgur.com/7Wy9Qow.gif
- https://github.com/OpenRCT2/OpenRCT2/commit/1f3998909 (which is 0.2.3): https://i.imgur.com/qad5MBE.gif
- https://github.com/OpenRCT2/OpenRCT2/commit/40e22eba8 (0.2.4): https://i.imgur.com/hlZNSAM.gif
- https://github.com/OpenRCT2/OpenRCT2/commit/966ba6b01 (tip): https://i.imgur.com/3UKZBC9.gif

It seems to be working fine, since August. I didn't see the off by one cutting and the mowing seemed to always seem the same thing

**Comment by Gymnasiast** [comment]
@tupaschoal All of the GIFs you posted are showing the handyman moving in the wrong direction, though. He is mowing lanes from bottom-left to top-right, while the lanes that appears are from top-left to bottom-right, so rotated by 90°.

**Comment by tupaschoal** [comment]
Yes, that I noticed, but I went as far as January 1st, 2019 and still saw the exact same behavior, so I started wondering whether it had always been a bug

**Comment by Gymnasiast** [comment]
Oh, some bugs have been in the game for years. Some are even the result of a typo that was made during initial implementation. Still a bug, though.

**Comment by tupaschoal** [comment]
Yup, I started Vanilla and saw the behavior was different P

**Comment by Gymnasiast** [comment]
@AaronVanGeffen Why are labelling this `original bug`, while @tupaschoal says that vanilla behaves **differently**?

**Comment by tupaschoal** [comment]
I can add a gif later today, but basically I loaded the same save used for the gifs above and vanilla RCT2 has the sprites in the exact orientation as the handyman is cutting.

**Comment by Gymnasiast** [comment]
That's how I understood it, yes, which makes this a `bug`. It appears that Aaron misread it.

**Comment by IntelOrca** [comment]
Maybe we should rename it to "vanilla bug"

**Comment by AaronVanGeffen** [comment]
@Gymnasiast I did indeed -- apologies. :)

**Comment by Ruedii** [comment]
This might be a vanilla bug indeed.

It looks like something that would be considered in a modern game, but not in a game of it's time.

Clearly the kind of detail orientation we can add (having the Handymen always mow the grass in the same direction and make the grass show freshly mowed stripes in that direction.)

Considering how few players even have their handymen mow the grass, due to the insanely low cost-benefit of doing so, it makes sense that it was never noticed.

**Comment by duncanspumpkin** [comment]
I just removing all the discussion about it being a vanilla bug as we have established that it is not a vanilla bug and is a mistake that has been around for a while.

**Comment by duncanspumpkin** [comment]
It seems to be a display issue caused by Terrain Surface objects refactor.

**Comment by duncanspumpkin** [comment]
Looks to be a bug in our object. Unfortunately CS layed out the images in an odd order. It goes:

rotation | base_image | old_set
----|----| ------
0 | 0xA8D | 2
1 | 0xA67 | 0
2 | 0xA8D | 2
3 | 0xA67 | 0

In our object format we have the following:

rotation |  new_set
----| ------
0 | 3
1 | 5
2 | 3
3 | 5

Where a new_set is converted into the old set by taking away 3.

So all we need to do to fix it is reorder the special section in our object for terrain_surface.
